src/noInapplicableLifeCycleRule.ts

```diff
@@ -0,0 +1,109 @@
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+import * as SyntaxKind from './util/syntaxKind';
+import {sprintf} from 'sprintf-js';
+import {Ng2Walker} from './angular/ng2Walker';
+import {difference, intersection} from './util/setOperations';
+
+
+function getInterfaceNames(t: any): string {
+  if (t.expression && t.expression.name) {
+    return t.expression.name.text;
+  }
+  return t.expression.text;
+}
+
+function getInterfaces(node: ts.ClassDeclaration): any[] {
```

Comment:
So if your example was just he first line, it would fail because `A` is not a component.

But _becase_ it's extended, it's fine to have a class which implements `OnInit`. So as long as it's used somewhere _down_ the inheritence chain, implementing is not a lint error.

Am I following?